apps/src/templates/instructions/TeacherFeedback.jsx

```diff
@@ -37,9 +38,60 @@ const styles = {
 
 class TeacherFeedback extends Component {
   static propTypes = {
-    viewAs: PropTypes.oneOf(['Teacher', 'Student']),
     //temp prop for which version to display (stable, released 2018-teacher-experience, or internal, developer version)
-    withUnreleasedFeatures: PropTypes.bool
+    withUnreleasedFeatures: PropTypes.bool,
+
+    //Provided by Redux
+    viewAs: PropTypes.oneOf(['Teacher', 'Student']),
+    serverLevelId: PropTypes.number,
+    teacher: PropTypes.number,
+  };
+
+  constructor(props) {
+    super(props);
+    const search = window.location.search;
+    const studentId = search.split('&')[1].split("=")[1];
+
+    this.state = {
+      comment: "",
+      studentId: studentId,
+      latestFeedback: [],
+    };
+  }
+
+  componentDidMount = () => {
+    $.ajax({
+      url: '/api/v1/teacher_feedbacks/get_feedback_from_teacher?student_id='+this.state.studentId+'&level_id='+this.props.serverLevelId+'&teacher_id='+this.props.teacher,
+      method: 'GET',
+      contentType: 'application/json;charset=UTF-8',
+    }).done(data => {
+      this.setState({latestFeedback: [data]});
+    });
+  };
+
+  onCommentChange = (event) => {
+    this.setState({comment: event.target.value});
+  };
+
+  onSubmitFeedback = () => {
+    const payload = {
+      comment: this.state.comment,
+      student_id: this.state.studentId,
+      level_id: this.props.serverLevelId,
+      teacher_id: this.props.teacher
+    };
+
+    $.ajax({
+      url: '/api/v1/teacher_feedbacks',
+      method: 'POST',
+      contentType: 'application/json;charset=UTF-8',
+      dataType: 'json',
+      data: JSON.stringify({teacher_feedback: payload})
+    }).done(data => {
+      this.setState({latestFeedback: [data]});
```

Comment:
Should the comment state (and textarea) be cleared when submission succeeds?

Reply:
Good question! The spec asks for being able to edit the text in the comment box and save again. I'll probably leave this part in for now, but if we get feedback that it's annoying/confusing, then it'll make sense to change it.